### PR TITLE
Set minimum supported julia version to 1.6 (LTS)

### PR DESCRIPTION
**Pull request by pabloferz**

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -30,11 +30,6 @@ jobs:
             arch: x86
           - os: windows-latest
             arch: x86
-        include:
-          # Add a 1.5 job because that's what Invenia actually uses
```
Comment (pabloferz) [inline .github/workflows/CI.yml:34]: Is this still true?

**Review by omus — COMMENTED** [review]
Changes look good. The 32-bit failures appear unrelated but should be sorted out

**Comment by omus** [comment]
🎉  Very nice work! I suggest we should also increase the minimum version of Julia in the Project.toml to be `1.6` as we no longer support earlier versions and don't want people accidentally installing this on older Julia revisions. Besides that I'm happy to approve.

Hopefully, someone from Invenia can validate the unresolved discussion.

**Review by omus — APPROVED** [review]

**Comment by omus** [comment]
I'll merge this by EOD unless there are objections. If Invenia wants to keep testing against Julia 1.5 we can always add that back in.

**Pull request merged**